src/SpotifyRemote.ts

```diff
@@ -76,7 +75,11 @@ interface SpotifyRemoteApiExtensions {
  * @export
  * @interface SpotifyRemoteApi
  */
-export interface SpotifyRemoteApi extends TypedEventEmitter<SpotifyRemoteEvents>, SpotifyRemoteApiExtensions {
+export interface SpotifyRemoteApi extends SpotifyRemoteApiExtensions {
+    /**
+     * React Native NativeEventEmitter
+     */
+    events: TypedEventEmitter<SpotifyRemoteEvents>,
```

Comment:
Since we have this javascript / typescript wrapper around the native module, it gives us flexibility to reduce the scope of this breaking change. 

Based on the definition here: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b2d66e0e9ff8d2e90134ba0d3e30a9edd3133a9c/types/react-native/index.d.ts#L178

I created a gist here: https://gist.github.com/cjam/2696db507999d514c2fb0c0674bfc207

That I think could help us preserve more of the existing interface and make the transition to your changes more seamless.  Essentially it just places the methods of the NativeEventEmitter onto the SpotifyRemote object when it gets imported.

Then in the SpotifyRemote.ts we could do something like this:

```ts
// SpotifyRemote.ts

/**
 * @ignore
 */
const nativeEventEmitter = new NativeEventEmitter(NativeModules.RNSpotifyRemoteAppRemote);

/** @ignore */
bindNativeEmitterEvents(SpotifyRemote, nativeEventEmitter);
```

Reply:
Thanks for writing this up! As mentioned in my other comment, I concern about rewriting the `addListener` function for the native module as it is required as part of the native `NativeEventEmitter` implementation (a `ReactMethod`): https://reactnative.dev/docs/native-modules-android#sending-events-to-javascript. Would rewriting it break something?

Reply:
Without that concern, it would better both way too since we also can restore the previous lazy subscription/unsubscription JS logic if we have our custom addListener/removeListener functions (which calls the native one  also keep track of listener registering at the same time)

Reply:
Yea good points.  I actually think it should all work though because we aren't actually overriding the `addEventListener` on the `NativeModules.RNSpotifyRemoteAppRemote` which is the native module that get's bound  to the `NativeEventEmitter`, we're actually overriding the `addEventListener` on the publicly exposed interface `SpotifyRemote`.  So the `NativeEventEmitter` should work fine and we just provide a thin wrapper around  it in the `SpotifyRemote` class / object.

Reply:
But the publicly exposed interface SpotifyRemote is `NativeModules.RNSpotifyRemoteAppRemote` atm (https://github.com/cjam/react-native-spotify-remote/blob/master/src/SpotifyRemote.ts#L259). I did try to override it anyway, and it indeed stops working. Will try to create an actual wrapper for it.

Reply:
I added a wrapper and was able to restore some of the API so we can avoid breaking changes.
https://github.com/cjam/react-native-spotify-remote/pull/189/commits/fbc6b4a88d98900b4493e605ea6bcc5301ac9ef0